### ISSUE TITLE
Possible bug with middleware typings

### ISSUE THREAD

**Issue by arackaf** [tannerlinsley/react-location#3060]
### Which project does this relate to?

Router

### Describe the bug

Possibly invalid type error

### Your Example Website or App

https://github.com/arackaf/tanstack-start-blog-dataloading/tree/special/start-bug-1

### Steps to Reproduce the Bug or Issue

This repo https://github.com/arackaf/tanstack-start-blog-dataloading/tree/special/start-bug-1 -- note the branch! special/start-bug-1
npm i

open routes/app/epics/$epicId/edit.tsx

Note the middleware - lines 20 and 40

Basically I'm trying to add some context on the server, AFTER the serverFn runs, and then read that context on the client. Both error out with TS, currently. Not sure if it's a bug, or a skill issue on my part :)   

### Expected behavior

no errors 

### Screenshots or Videos

n/a

### Platform

n/a

**Comment by schiller-manuel** [comment]
is this still valid?

**Comment by arackaf** [comment]
Yeah - I've spoken to @tannerlinsley many times about this one - getting this fixed is the last thing y'all need before single-flight mutations will work

**Comment by LadyBluenotes** [comment]
I believe this was fixed by the middleware context typing updates in #3375. Server-function middleware context now should aligns with the runtime behaviour.

If you are still seeing a type error on a current version, please open a new issue with a minimal repro.
